apps/language_server/test/server_test.exs

```diff
@@ -336,4 +336,56 @@ defmodule ElixirLS.LanguageServer.ServerTest do
              ]) = resp
     end)
   end
+
+  test "loading of umbrella app dependencies", %{server: server} do
+    in_fixture(__DIR__, "umbrella", fn ->
+      # We test this by opening the umbrella project twice.
+      # First to compile the applications and build the cache.
+      # Second time to see if loads modules
+      initialize(server)
+
+      # Upon first vist, server complies and loads all umbrella applications and modules
+      Process.sleep(1_500)
```

Comment:
If the server process is handling a message, then simply calling `:sys.get_state` will make the calling process block until the server process is ready. If the server is waiting for an async message from other process you can call `:sys.get_state` in a loop with short sleeps until the state returned indicates that async task is completed.

Reply:
ok. I'll will try to update test

Reply:
updated, please check

Reply:
Looks good